### ISSUE TITLE
Minifying concatenates strings strangely

### ISSUE THREAD

**Issue by nebrelbug** [terser/terser#592]
**Bug report or Feature request?**
Feature Request

**Version (complete output of `terser -V` or specific git commit)**
Presumably latest?

**Complete CLI command or `minify()` options used**
Used https://try.terser.org

**`terser` input**

```js
function x (){var tR='';tR+='Test String';console.log("Console log");return tR}
```

**`terser` output or error**
```js
function x(){return"Test String",console.log("Console log"),"Test String"}
```

**Expected result**
```js
function x(){console.log("Console log");return"Test String"}
```
Or even:
```js
function x(){return"",console.log("Console log"),"Test String"}
```

Terser's output seems redundant... am I missing an API option?

**Comment by L2jLiga** [comment]
Does extra passes solve problem?

**Comment by fabiosantoscode** [comment]
Yes. The reason terser is multi-pass is that sometimes a single pass can't fix all the things.

That said, I'm always looking for easy to implement and fast ways to optimise as most as we can in one pass. 2 or 3 passes should only be used in corner cases.

**Comment by nebrelbug** [comment]
Is there an API option to run multiple passes?

**Comment by L2jLiga** [comment]
Here's 'passes' option in compress
